src/main/java/edu/gatech/cs2340/trydent/OneTimeEvent.java

```diff
@@ -0,0 +1,75 @@
+package edu.gatech.cs2340.trydent;
+
+
+/**
+ * Creates an event that starts, updates, and stops exactly once.
+ *
+ */
+public abstract class OneTimeEvent {
+    private ContinuousEvent event;
+
+    public OneTimeEvent() {
+        event = new ContinuousEvent(){
+            @Override
+            public void onStart(){
+                OneTimeEvent.this.onStart();
+            }
+
+            @Override
+            public void onUpdate() {
+                OneTimeEvent.this.onUpdate();
+                this.stop();
+            }
+
+            @Override
+            public void onStop(){
+                OneTimeEvent.this.onStop();
+            }
+        };
+        TrydentEngine.addContinuousEvent(event);
```

Comment:
This line is unnecessary, because the constructor of ContinuousEvent does this automatically. Thus won't hurt, since the engine stores continuous events in a Set (so duplicate events are ignored); it's just extraneous and we should avoid it in our internal code.